pkg/controller/atlasproject/maintenancewindow.go

```diff
@@ -0,0 +1,157 @@
+package atlasproject
+
+import (
+	"context"
+	"errors"
+
+	"go.mongodb.org/atlas/mongodbatlas"
+
+	mdbv1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/project"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
+)
+
+// ensureMaintenanceWindow ensures that the state of the Atlas Maintenance Window matches the
+// state of the Maintenance Window specified in the project CR. If a Maintenance Window exists
+// in Atlas but is not specified in the CR, it is deleted.
+func ensureMaintenanceWindow(ctx *workflow.Context, projectID string, atlasProject *mdbv1.AtlasProject) workflow.Result {
+	windowSpec := atlasProject.Spec.MaintenanceWindow
+	if err := validateMaintenanceWindow(windowSpec); err != nil {
+		return workflow.Terminate(workflow.ProjectWindowInvalid, err.Error())
+	}
+
+	if isEmptyWindow(windowSpec) {
+		ctx.Log.Debugw("Window empty or undefined, deleting in Atlas")
+		if result := deleteInAtlas(ctx.Client, projectID); !result.IsOk() {
```

Comment:
This will send `DELETE` request each reconcile call.
We can send delete only if `status` is not empty which `spec` is empty.

Reply:
The function `isEmptyWindow` will return true only if none of the fields are specified. In that case we will indeed delete the window.
Otherwise no delete request should be sent.

Do you think the behaviour should be different ?

Reply:
Default project has no maintenance window defined so `isEmptyWindow` will return `true` and we will make the `DELETE` request on each call of Reconcile. This isn't critical but it is possible to avoid it by setting and checking the status for the maintenance window.

Just a suggestion though =)

Reply:
We can merge my PR with conditions removal logic. You will have to rebase but afterwards you will be able to use the presence of condition for determining if you need to delete anything in Atlas.